test/test_formatter.py

```diff
@@ -0,0 +1,114 @@
+import pytest
```

Comment:
I think there are two issues with the encoding:

1) On python 2, it's failing because you're using unicode characters in literal strings. I *think* if you add `from __future__ import unicode_literals` to the very top of your file you'll stop seeing the SyntaxError.

2) The assertion is failing in python 3 (and will likely also fail in python 2 once you fix the first issue). I think this is because the literal is a `string` type, but the value that your computing from the formatter is `bytes`. Try doing something like this:

    all_formatters['table'](test_data).decode('utf8', 'replace')

If that doesn't work, try printing the types of both values:

    print(type(formatter...))
    print(type(expected_table))

They'll need to match up and that will help us figure out why it's failing.

Reply:
Hi @ahal , I tried to remove py2 from tox.ini file, and everything works. I don't know why py2 cause assertion errors!

Reply:
Hm, that's really strange! I'll pull this down and poke around a bit when I get a chance. Maybe we should just drop python 2 support anyway :)

Reply:
Did you make any changes to this PR? I pulled this down and removed the python 2 from tox.ini, but the tests still fail for me afterwards.

Maybe we should move this into data files rather than storing them in the test file. That should help fix the python 2 errors. Then the python 3 errors should just be a matter of decoding/encoding both sets of strings properly.

Reply:
I didn't change the file recently, I rechecked tox, see the screen below:
![](https://i.imgur.com/Pbc0JNn.png)

I will work on separating data from tests.

Reply:
Thanks!

One other issue I'll mention is that the test names have the entire unicode in them. We should try to make this more legible (just a number for might be better). This page has some tips on how you can accomplish this:
https://docs.pytest.org/en/latest/example/parametrize.html#different-options-for-test-ids

Reply:
I am working on this, first, separated the test data into json files for each formater.  tempdir fixture was used to read the files, then, saved the data into a list. Now, I want to pass as a parameter to another fixture, which will pass each fomatter test cases to a test function, still couldn't make it, but I will keep trying.

Reply:
hi @ahal , the problem still exists even after putting the test data outside. I think it is a bug in tox, python 3 works fine but not py2,  please, have a look at tox log with only py 3 and py2+py3
https://gist.github.com/nadaa/432eb32c9efd0074f9608929b6233b0d

Reply:
Ok thanks for the attempt, I'll try again in a bit (I've been out most of the week so haven't much time to investigate too far yet).

Reply:
You're welcome. Shall I push the update?

Reply:
Yes please!